### PR TITLE
[FLINK-31733][docs] Fix/Detect OpenAPI model name clashes 

### PR DESCRIPTION
**Pull request by zentol**
Fixes 5 cases where a name clash caused certain classes to not be documented properly. Name clashes were silently ignored by the generator.

The offending classes are now annotated to provide a unique name, and the generator was enhanced to detect such clashes.

### CODE REVIEW

File: flink-docs/src/main/java/org/apache/flink/docs/rest/OpenApiSpecGenerator.java [hunk 1]
```diff
@@ -494,4 +498,30 @@ private static PathItem.HttpMethod convert(HttpMethodWrapper wrapper) {
         }
         throw new IllegalArgumentException("not supported");
     }
+
+    /** A {@link TypeNameResolver} that detects name-clashes between top-level and inner classes. */
+    public static class NameClashDetectingTypeNameResolver extends TypeNameResolver {
```
Comment (dmvk) [inline flink-docs/src/main/java/org/apache/flink/docs/rest/OpenApiSpecGenerator.java:503]: This is nice. <3

File: flink-runtime/src/main/java/org/apache/flink/runtime/rest/messages/job/JobDetailsInfo.java [hunk 2]
```diff
@@ -259,6 +261,7 @@ public String getJsonPlan() {
     // ---------------------------------------------------
 
     /** Detailed information about a job vertex. */
+    @Schema(name = "JobDetailsVertexInfo")
     public static final class JobVertexDetailsInfo {
```
Comment (dmvk) [inline flink-runtime/src/main/java/org/apache/flink/runtime/rest/messages/job/JobDetailsInfo.java:265]: Why don't we just rename the class? do we need to keep any bw compatibility on this level?

File: flink-runtime/src/test/java/org/apache/flink/runtime/rest/util/DocumentingRestEndpoint.java [hunk 3]
```diff
@@ -50,4 +51,11 @@ List<Tuple2<RestHandlerSpecification, ChannelInboundHandler>> initializeHandlers
                                         spec2.getTargetRestEndpointURL()))
                 .collect(Collectors.toList());
     }
+
+    static DocumentingRestEndpoint forRestHandlerSpecifications(RestHandlerSpecification... specs) {
+        return localAddressFuture ->
+                Arrays.asList(specs).stream()
```
Comment (dmvk) [inline flink-runtime/src/test/java/org/apache/flink/runtime/rest/util/DocumentingRestEndpoint.java:57]:
```suggestion
                Arrays.stream(specs)
```

File: flink-runtime/src/main/java/org/apache/flink/runtime/rest/messages/job/JobDetailsInfo.java (hunk 2 above)
Comment (zentol) [reply flink-runtime/src/main/java/org/apache/flink/runtime/rest/messages/job/JobDetailsInfo.java:265]: We could rename all classes, I just wasn't sure if we want to go down that route purely for documentation work.

Comment (zentol) [reply flink-runtime/src/main/java/org/apache/flink/runtime/rest/messages/job/JobDetailsInfo.java:265]: Compatibility isn't a concern.

Comment (zentol) [reply flink-runtime/src/main/java/org/apache/flink/runtime/rest/messages/job/JobDetailsInfo.java:265]: There are of course advantages to renaming the class, like it being easier to track down the internal representation of something in the API. But I'm not sure if that's worth the hassle of restricting us to unique class names everywhere.

**Review by dmvk — APPROVED** [review]
👍

File: flink-runtime/src/main/java/org/apache/flink/runtime/rest/messages/checkpoints/CheckpointingStatistics.java [hunk 4]
```diff
@@ -198,6 +200,7 @@ public int hashCode() {
     }
 
     /** Checkpoint summary. */
+    @Schema(name = "CheckpointStatisticsSummary")
```
[the remark this answers is not in the document]
Comment (zentol) [reply flink-runtime/src/main/java/org/apache/flink/runtime/rest/messages/checkpoints/CheckpointingStatistics.java:203]: That's what I tried first, but the result wasn't satisfying. `JobDetailsInfoJobVertexDetailsInfo` isn't a great name :/

[the remark this answers is not in the document]
Comment (zentol) [reply flink-runtime/src/main/java/org/apache/flink/runtime/rest/messages/checkpoints/CheckpointingStatistics.java:203]: It's possible to do though in the custom `TypeNameResolver`.

**Pull request merged**